Scenes/CharacterCreation/CharacterCreation.gd

```diff
@@ -71,59 +81,44 @@ func _on_Name_text_changed():
-		Player.inventory.add(axe, 1)
-		Player.equip(axe)
-		Player.attacks["melee"].append(4)
-	if chosenEquip == "Quick Shooter":
-		var bow = Classes.CreateItem("weapons", "ranged", "hunting bow", "Bow", [0,0,0,0,0,2,0,0,0,0], 1)
-		var arrow = Classes.CreateItem("weapons", "consumables", "arrow", "Arrow", [0,0,0,0,0,0,1,5,0,0], 1)
-		var dagger = Classes.CreateItem("weapons", "melee", "dagger", "Dagger", [1,0,0,0,0,0,0,5,0,0], 1)
-		Player.inventory.add(bow, 1)
-		Player.equip(bow)
-		Player.inventory.add(arrow, 10)
-		Player.equip(arrow)
-		Player.inventory.add(dagger, 1)
-		Player.equip(arrow)
-		Player.attacks["ranged"].append(1)
-		Player.attacks["melee"].append(4)
-	
-	#load default save data and override player with chosen data then save the file and start the game
-	var data = {
-		name = Player.name,
-		info = "Breathes fire",
-		picture = Player.pic[0],
-		saveFile = Player.name + " - "+ str(OS.get_unix_time()),
-		player = Player
-	}
-	SaveGame.save_character(data)
-	var Player2 = Classes.CreateCharacter("Alrune", [8,8,8,8,8,8,8], "res://Assets/Images/Profiles/Friendlies/Tex_AnimeAva_28.png", [true, "res://Assets/Images/Profiles/ImageBorder.png"], [[1, 2, 3, 4], [1], [1]], 1, [0], 5, 2, 200, 200)
-	var Enemy = Classes.DeathHound([[], [], []], 14, [0], 1, 0.5, 100)
-	var Enemy2 = Classes.DeathHound([[], [], []], 15, [0], 1, 0.5, 100)
-	var Enemy3 = Classes.DeathHound([[], [], []], 16, [0], 1, 0.5, 100)
-	Enemy3.health = 0
-	Enemy3.name = "Dead Hound"
-	#print (str(Enemy.stats))
-	#print (str(Enemy2.stats))
-	#print (str(Enemy3.stats))
-	var BattleSceneMaker = preload("res://Scenes/BattleScenes/Create_Battle.gd")
-	BattleSceneMaker.switchScene(Core, "Wolf Den", "res://Assets/Images/Backgrounds/Forest.jpg", [Player, Player2], [Enemy, Enemy2, Enemy3])
+	
+	var player = CharacterManager.create(character_names[selected_character])
+	player.nickname = character_name
+	player.stats = stats
+	
+	var chosen_equip = CharacterDefaults.starting_equipment.keys()[selected_equip]
+	Core.emit_signal("msg", "Chosen equipment: " + chosen_equip, Core.INFO, self)
+	player.equipment = CharacterDefaults.starting_equipment[chosen_equip]
+	
+	player.file = player.meta.name + " - "+ str(OS.get_unix_time())
+	Core.player = player
+	
+	Core.emit_signal("request_scene_load", battle_scene)
+	var error = Core.connect("scene_loaded", self, "_on_scene_loaded")
+	if error:
+		Core.emit_signal("msg", "Event scene_loaded failed to bind", Core.WARN, self)
+	
+	Core.emit_signal("request_scene_load", battle_scene)
+
+func _on_scene_loaded(scene):
+	if scene != battle_scene:
+		return
+	
+	SaveGame.save_character(Core.player)
+	
+	var player2 = CharacterManager.create(character_names[selected_character])
+
+	var enemy = CharacterManager.create("death_hound")
+	var enemy2 = CharacterManager.create("death_hound")
+	var enemy3 = CharacterManager.create("death_hound")
```

Comment:
A level will need to be passed on creation to scale it

Reply:
Also, after being created any extra skills that may need to be added will be added seperately?

Reply:
So this can be done by the character management script?

Reply:
Yep!

Reply:
I'll save this for another PR!